src/ast/source-range.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+export interface SourceRange {
+  /* The resolved path to the file. */
+  file: string;
```

Comment:
shouldn't the file be implicit? A Feature should belong to a document, so if a Feature it should be possible to get SourceRange's file from it's Feature. IOW, we need something like `.document` on Features.